src/figuresClasses.ts

```diff
@@ -1,19 +1,105 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 export interface Figure {
-
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
-export class Circle {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('sides must be bigger than 0');
+    }
 
+    if (this.a > this.b && this.a > this.c && this.a >= (this.b + this.c)) {
+      throw new Error('sides 1, 2 and 3 cant form a triangle');
+    }
+
+    if (this.b > this.a && this.b > this.c && this.b >= (this.a + this.c)) {
+      throw new Error('sides 1, 2 and 3 cant form a triangle');
+    }
+
+    if (this.c > this.a && this.c > this.b && this.c >= (this.a + this.b)) {
+      throw new Error('sides 1, 2 and 3 cant form a triangle');
+    }
```

Comment:
```suggestion
    if (this.a >= (this.b + this.c) || this.b >= (this.a + this.c) || this.c >= (this.a + this.b)) {
      throw new Error('sides 1, 2 and 3 cant form a triangle');
    }
```
it can be shorter